### PR TITLE
Update k8s library dependencies to 1.12.0.-rc2

### PR DESCRIPTION
**Pull request by aanm**
As kubernetes is going to be released in a couple days we should update the libraries dependencies so that Cilium 1.3 is up to date with k8s 1.12.

Once 1.12 is released the remaining diff should be backported although I wouldn't expect to be that much of a difference.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5682)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
I need to check failure of https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/6710/testReport/junit/k8s-1/8/K8sTunnelTest_VXLan_Check_VXLAN_mode/

**Comment by aanm** [comment]
I need to check failure of https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/6710/testReport/junit/k8s-1/8/K8sTunnelTest_VXLan_Check_VXLAN_mode/

**Comment by aanm** [comment]
test-me-please

File: contrib/scripts/check-fmt.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ diff="$(find . ! \( -path './contrib' -prune \) \
 if [ -n "$diff" ]; then
 	echo "Unformatted Go source code:"
 	echo "$diff"
-	exit 1
```
Comment (ianvernon) [inline contrib/scripts/check-fmt.sh]: Was this intended to be kept in the PR?

**Review by ianvernon — APPROVED** [review]

**Review by tgraf — APPROVED** [review]
Looks good aside from what @ianvernon pointed out

File: contrib/scripts/check-fmt.sh (hunk 1 above)
Comment (aanm) [reply contrib/scripts/check-fmt.sh]: No, but it should :-)

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by jrajahalme — APPROVED** [review]

**Pull request merged**